proposed/2021/MachineReadableOutput.md

```diff
@@ -0,0 +1,797 @@
+   > Microsoft.Extensions.Primitives      [1.0.0, 5.0.0]        1.0.0
+   > NuGet.Commands                       4.8.0-preview3.5278   4.8.0-preview3.5278
+   > Text2Xml.Lib                         [1.1.2, 2.0.0)        1.1.2
+
+   Transitive Package                                                                   Resolved
+   > Microsoft.CSharp                                                                   4.0.1
+   > Microsoft.NETCore.Platforms                                                        1.1.0
+   > Microsoft.NETCore.Targets                                                          1.1.0
+...
+
+Project 'MyProjectB' has the following package references
+   [netcoreapp3.1]:
+   Top-level Package      Requested             Resolved
+   > NuGet.Commands       4.8.0-preview3.5278   4.8.0-preview3.5278
+   > Text2Xml.Lib         1.1.2                 1.1.2
+
+   Transitive Package                                                                   Resolved
+   > Microsoft.CSharp                                                                   4.0.1
+   > Microsoft.NETCore.Platforms                                                        1.1.0
+   > Microsoft.NETCore.Targets                                                          1.1.0
+   > Microsoft.Win32.Primitives                                                         4.3.0
+...
+
+   [net5.0]:
+   Top-level Package      Requested             Resolved
+   > NuGet.Commands       4.8.0-preview3.5278   4.8.0-preview3.5278
+   > Text2Xml.Lib         1.1.2                 1.1.2
+
+   Transitive Package                                                                   Resolved
+   > Microsoft.CSharp                                                                   4.0.1
+   > Microsoft.NETCore.Platforms                                                        1.1.0
+   > Microsoft.NETCore.Targets                                                          1.1.0
+...
+
+```
+
+### dotnet list package --include-transitive --json --flatten
+
+```json
+{
+  "version": 1,
+  "topLevelPackages": [
+       {
+            "id": "Microsoft.Extensions.Primitives",
+            "requestedVersion": "[1.0.0, 5.0.0]",
+            "resolvedVersion": "1.0.0"
+        },
+        {
+            "id": "NuGet.Commands",
+            "requestedVersion": "4.8.0-preview3.5278",
+            "resolvedVersion": "4.8.0-preview3.5278"
+        },
+        {
+            "id": "Text2Xml.Lib",
+            "requestedVersion": "[1.1.2, 2.0.0)",
+            "resolvedVersion": "1.1.2"
+        }
+    ],
+    "transitivePackages": [
+        {
+            "id": "Microsoft.CSharp",
+            "resolvedVersion": "4.0.1"
+        },
+        {
+            "id": "Microsoft.NETCore.Platforms",
+            "resolvedVersion": "1.1.0"
+        },
+        {
+            "id": "Microsoft.NETCore.Targets",
+            "resolvedVersion": "1.1.0"
+        }
+...
+    ]
+}
+```
+
+### Compatibility
```

Comment:
I disagree. Interactive tools can change output because they're designed for humans to read, and when it unexpectedly changes output, humans can adapt. Adding a json output makes it an API, even if it's being output from a CLI tool rather than a web API.

Also consider the fact that it's most likely going to be used in CI scripts, and the version of the .NET SDK installed on the CI agent might not be in control of the team whose product is being built, it might be controlled by an infrastructure team, or another company altogether (GitHub and Azure DevOps hosted agents for non-Microsoft employees).

Reply:
I still don't see how this is beneficial. The SDK version should be used to infer compatibility, not some command specific output "version". I haven't seen this in other CLI experiences output either. Why should we do this?